lib/adbc_helper.ex

```diff
@@ -289,4 +297,25 @@ defmodule Adbc.Helper do
         "#{:code.priv_dir(:adbc)}/lib"
     end
   end
+
+  defp adbc_cache_dir() do
+    dir =
+      if dir = System.get_env("ADBC_CACHE_DIR") do
+        Path.expand(dir)
+      else
+        :filename.basedir(:user_cache, "adbc")
+      end
+
+    if File.dir?(dir) do
+      {:ok, dir}
+    else
+      r = File.mkdir_p(dir)
+
+      if r == :ok do
+        {:ok, dir}
+      else
+        r
+      end
+    end
```

Comment:
I wonder if we should just return the path as it will return error later on anyway?

```suggestion
    if dir = System.get_env("ADBC_CACHE_DIR") do
      Path.expand(dir)
    else
      :filename.basedir(:user_cache, "adbc")
    end
```

Reply:
I see. So i think we can do this:

```suggestion
    dir =
      if dir = System.get_env("ADBC_CACHE_DIR") do
        Path.expand(dir)
      else
        :filename.basedir(:user_cache, "adbc")
      end

    File.mkdir_p(dir)
```

Just slightly cleaner.

Reply:
Wait. But we handle the error of `File.write`, which is why I don't think we need to handle it here.

Reply:
But `File.mkdir_p` only returns `:ok` | `{:error, reason}`, we need to return `{:ok, dir}` if it succeeds. So maybe

```suggestion
    dir =
      if dir = System.get_env("ADBC_CACHE_DIR") do
        Path.expand(dir)
      else
        :filename.basedir(:user_cache, "adbc")
      end

    case File.mkdir_p(dir) do
      :ok ->
        {:ok, dir}

      other ->
        other
    end
```